### DISCUSSION TITLE
Support for glTF models?

### DISCUSSION THREAD

**Discussion by mattiaverga** [CelestiaProject/Celestia#2154] [Ideas]
NASA/JPL provides 3d models of many minor bodies in glTF format. As far as I understand, these models provides everything needed to render the object, without the needs of external textures or normal maps.
An example of Hyperion is viewable [here](https://science.nasa.gov/saturn/moons/hyperion/) and the model is downloadable [here](https://science.nasa.gov/resource/hyperion-3d-model/). The NASA model seems a bit more detailed than the current one in Celestia.

Would be worthwhile to add support for glTF models in Celestia and switch bodies to use those 3d models instead of combining texture map, normal map and cmod model? That would also solve licensing issues.

**Comment by 375gnu** [comment]
glTF is much more complicated format than one supported by Celestia. So this should be an item for the next version where glTF should be default format and CMOD/3DS should we deprecated.

**Reply by Mapoet** [reply]
If it could support integrating orbit for user-defined system，more people may be intersited in simulating N-Body paths.
